### ISSUE TITLE
Umbraco 8.6.3 > Doesn't Work

### ISSUE THREAD

**Issue by jason-delaplain** [callumbwhyte/friendly-sitemap#9]
I've installed the package by Our.Umbraco, GitHub and tried Local install, and it doesn't do anything. I don't see a "sitemap.xml" file, nor does going to the url "/sitemap.xml" show anything (it's 404 error).  There isn't much instruction on what else to do, if I'm supposed to.  I've tried publishing a page, building the project, ensuring IISExpress is stopped and re-running the project, etc.

**Comment by callumbwhyte** [comment]
Hey @deladroid!

Thanks for trying Friendly Sitemap! Sorry to hear you've had issues getting things up and running...

The sitemap is a virtual sitemap and therefore you won't see a physical sitemap.xml file on disk, but you should be able to browse to /sitemap.xml and see it.

In order to bind /sitemap.xml to a site the root node needs to have a domain assigned. Can you confirm you have a hostname assigned to your site root under the "Culture & Hostnames" menu?

Cheers,
Callum

**Comment by callumbwhyte** [comment]
Hey @deladroid,

I've had a brainwave over the weekend about what this issue might be!

The sitemap needs to be enabled via an app setting in your project's `web.config` file in order to work.

```
<add key="Umbraco.Sitemap.EnableSitemap" value="true" />
```

When installing via [NuGet](https://www.nuget.org/packages/Our.Umbraco.FriendlySitemap/) this is done automatically for you, but if you have installed via Our Umbraco or the backoffice you need to do this manually yourself.

If you can, I suggest installing the package via NuGet. If not, you should be able to add the above line to your `web.config` file.

Please let me know if this resolves your issue! 😊

Cheers,
Callum

**Comment by jason-delaplain** [comment]
Thanks, I'll try this soon, and report back.

**Comment by rongrisnich** [comment]
I'm facing the same issue, running Umbraco 8.5.4 and version 1.1.1 of your package. Unfortunately adding the key to the project's web.config file did not resolve the issue. /sitemap.xml returns a 404.

**Comment by john-mckillip** [comment]
I have this package installed and working on Umbraco 8.6.4. Make sure you have a domain set in Culture and Hostnames
 for the start node
![sitemap](https://user-images.githubusercontent.com/7633040/92489727-b8cc6900-f1b5-11ea-8175-cc2dc83e83e8.JPG)

**Comment by rongrisnich** [comment]
Tried with 8.6.4 just now, ensured domain is set on the root node and added the key Umbraco.Sitemap.EnableSitemap to the project's web.config file (set to true). Still no luck (404)... :(

**Comment by callumbwhyte** [comment]
Hey @rongrisnich @john-mckillip!

Thanks for trying out Friendly Sitemap – really sorry you're having issues.

Can you both confirm for me how the package was installed? Was this via NuGet? Or Our Umbraco / the Umbraco backoffice?

Can you both further confirm if there is a DLL / reference in your projects for `Our.Umbraco.Extensions.Routing`?

Cheers,
Callum

**Comment by rongrisnich** [comment]
I installed the package using the Umbraco backoffice.

Our.Umbraco.Extensions.Routing.dll is **not** there. Our.Umbraco.FriendlySitemap.dll is added to the root of the web application.

**Comment by callumbwhyte** [comment]
Aha! Thanks @rongrisnich!

Looks like this is the issue... For some reason the Umbraco package zip does not include that DLL, which is required to configure the routing. I will fix this and release a patch shortly.

In the mean time you can [download the DLL here](https://github.com/callumbwhyte/umbraco-routing-extensions/releases/download/release-1.0.0/Our.Umbraco.Extensions.Routing.v1.0.0.zip) and manually add it to the `bin` folder of your project to hopefully resolve your issue.

Cheers,
Callum

**Comment by rongrisnich** [comment]
Downloaded Our.Umbraco.Extensions.Routing.dll and added it to the `bin` folder, but unfortunately no luck yet...

**Comment by callumbwhyte** [comment]
Just to confirm @rongrisnich, the `bin` folder should now have both `Our.Umbraco.FriendlySitemap` and `Our.Umbraco.Extensions.Routing`... Do you still have the same error?

**Comment by rongrisnich** [comment]
Strangely, the file `Our.Umbraco.FriendlySitemap.dll` was placed in the root folder (not `bin`). Moving this file into the `bin` folder resolves the issue!

**Comment by callumbwhyte** [comment]
Fantastic, glad we found a way to get it to work.

I know what the issue is now and will fix accordingly 😊

**Comment by bzuidgeest** [comment]
I just came to this today, with the same issue. I have to note that indeed the dll's were in the root of the site and not the bin folder and the routing dll was missing.
I would like to add that this also happens to the friendly robot package. Don't forget to update that one.
For the record, I installed from the package dashboard in umbraco. 8.6.3. Once copied everything worked.

**Comment by simonwatt** [comment]
Same issue for me after installing via packages section on dashboard, but resolved after doing 4 things mentioned above (not sure if all steps were needed or not)

1) `<add key="Umbraco.Sitemap.EnableSitemap" value="true" />` in web.config
2) Moved Our.Umbraco.FriendlySitemap.dll from root folder into bin
3) Downloaded the Our.Umbraco.Extensions.Routing.dll linked to above, copy to bin folder
4) Added my domain to 'Culture and Hostnames' via 'root node' > 'actions' > 'culture and hostnames' on the umbraco dashboard.

**Comment by jkulker** [comment]
not sure what i am doing wrong did above 4 steps still no luck

**Comment by jason-delaplain** [comment]
Sorry for such a delay in me replying.  I put a domain in as stated above, and it's working (installed via Umbraco "Packages").  
- Confirm that Our.Umbraco.FriendlySitemap.DLL is in bin
- Confirm `<add key="Umbraco.Sitemap.EnableSitemap" value="true" />` is in web.config
- Confirm Domain added at "Root > Culture and Hostnames" (Ie., local dev "https://localhost:port")

**Comment by callumbwhyte** [comment]
Hello everyone!

Thanks for your patience.

For anyone who was still running into issues with the Our Umbraco version of the package, the latest v1.2.0 is now released and should install without any issues like before!

Cheers,
Callum

**Comment by HolmesSoftware** [comment]
> Hey @deladroid!
> 
> Thanks for trying Friendly Sitemap! Sorry to hear you've had issues getting things up and running...
> 
> The sitemap is a virtual sitemap and therefore you won't see a physical sitemap.xml file on disk, but you should be able to browse to /sitemap.xml and see it.
> 
> In order to bind /sitemap.xml to a site the root node needs to have a domain assigned. Can you confirm you have a hostname assigned to your site root under the "Culture & Hostnames" menu?
> 
> Cheers,
> Callum

Can you add this to your documentation? Your docs just say to enable through the web config. It would have been super helpful if this was in the docs.

**Comment by callumbwhyte** [comment]
Hey @Bleeus,

Thanks for using Friendly Sitemap!

The [usage docs](https://github.com/callumbwhyte/friendly-sitemap#usage) were recently updated to cover what is required to setup the sitemap.

As of the latest release v1.2 a domain is not required in order to run. If you have Friendly Sitemap installed already, update the `Our.Umbraco.Extensions.Routing` dependency to the latest version and you should be golden.

Cheers,
Callum

**Comment by HolmesSoftware** [comment]
Just updating you with the steps we took in case they help you or anyone else.

We did a fresh Umbraco install (Umbraco version 8.14.1) using the Zip (Not NuGet) and installed FriendlySitemap from the Umbraco Packages section. The package manager installed version 1.2.1 of FriendlySitemap.

We added the  `<add key="Umbraco.Sitemap.Enable" value="true" />` to the web.config.

The sitemap would not show up until we added the domain to the Culture & Hostnames section.
